core/scripts/chaincli/handler/keeper.go

```diff
@@ -193,8 +215,32 @@ func (k *Keeper) approveFunds(ctx context.Context, registryAddr common.Address)
 	log.Println(registryAddr.Hex(), ": KeeperRegistry approved - ", helpers.ExplorerLink(k.cfg.ChainID, approveRegistryTx.Hash()))
 }
 
+func (k *Keeper) verifyContract(address common.Address, params ...string) {
+	// Change to the contracts directory where the hardhat.config.ts file is located
+	if err := k.changeToContractsDirectory(); err != nil {
+		log.Fatalf("failed to change to directory where the hardhat.config.ts file is located: %v", err)
+	}
+
+	// Append the address and params to the commandArgs slice
+	commandArgs := []string{address.String()}
+	commandArgs = append(commandArgs, params...)
+
+	// Format the command string with the commandArgs
+	command := fmt.Sprintf(
+		"NODE_HTTP_URL='%s' EXPLORER_API_KEY='%s' NETWORK_NAME='%s' pnpm hardhat verify --network env %s",
+		k.cfg.NodeHttpURL,
+		k.cfg.ExplorerAPIKey,
+		k.cfg.NetworkName,
+		strings.Join(commandArgs, " "),
+	)
+
+	if err := k.runCommand(command); err != nil {
+		log.Fatal(err)
```

Comment:
I'm skeptical we should actually error the script here... verification can be super finicky. I think we should log the error but proceed.